networkx/algorithms/cluster.py

```diff
@@ -247,6 +247,80 @@ def wt(u, v):
         yield (i, dtotal, dbidirectional, float(directed_triangles))
 
 
+@not_implemented_for("directed")
+@nx._dispatchable
+def all_triangles(G, nodes=None):
+    """
+    Yields all unique triangles in an undirected graph using ID assignments and set intersections.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        An undirected graph. Self-loops and multiple edges are ignored.
+
+    nodes : node, iterable of nodes, or None (default=None)
+        If a node or iterable of nodes is given, only triangles involving those nodes are yielded.
+        If None, yields all unique triangles in the graph.
+
+    Yields
+    -------
+    tuple
+        A tuple of three nodes forming a triangle (u, v, w), ordered based on assigned ID,
+        guaranteed to be unique.
+
+    Examples
+    --------
+    >>> G = nx.complete_graph(4)
+    >>> list(all_triangles(G))
+    [(0, 1, 2), (0, 1, 3), (0, 2, 3), (1, 2, 3)]
+    """
+    if nodes is None:
+        node_to_id = {node: i for i, node in enumerate(G.nodes())}
```

Comment:
You can just iterate G

```suggestion
        node_to_id = {node: i for i, node in enumerate(G)}
```